### PR TITLE
Javascript/javascript1/week2

### PR DESCRIPTION
**Pull request by hamedveloper**

### CODE REVIEW

File: javascript/javascript1/week2/exerciseStep3Week2.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    } else {
+        return firstname + ' ' + surname;
+    }
+}
+
+const fullName1 = getFullName('Hamed', 'Karachi', true);
+const fullName2 = getFullName('Arman', 'Naderi', false);
+
+console.log(fullName1);
+console.log(fullName2);
```
Comment (tjshamhu) [inline javascript/javascript1/week2/exerciseStep3Week2.js:19]: Wow, I like this, beautiful code

File: javascript/javascript1/week2/exerciseStep3Week2.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+const n = new Date().getDay();
+const weekday = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
+
+function getEventWeekday(i) {
+    if (i >= 7) {
+        return ((i % 7) + n) - 1;
+    } else {
+        return ((i + n) % 7) - 1;
+    }
+}
```
Comment (tjshamhu) [inline javascript/javascript1/week2/exerciseStep3Week2.js]: I think this function may be missing a few things. Seems like it's returning a number and not a weekday. Also, test with giving it an argument that is larger than 7

File: javascript/javascript1/week2/exerciseStep3Week2.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+
+
+//Candy helper//////////////////////////////////////////////////////////////////////////////
+
+const boughtCandyPrices = [];
+
+function addCandy(candyType, weight) {
+    if (candyType === 'Sweet') {
+        var pricePerGram = 0.5;
```
Comment (tjshamhu) [inline javascript/javascript1/week2/exerciseStep3Week2.js:110]: In most cases, you would avoid using `var`. It's good to know how it works, but rather stick to `const` and `let`

File: javascript/javascript1/week2/exerciseStep3Week2.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        return 'wear shorts and T-shirt'
+    }
+}
+
+//Student Manager/////////////////////////////////////////////////////////////////
+
+var class07Students = [];
+
+function addStudentToClass(studentName) {
+    if (studentName !== '') {
```
Comment (tjshamhu) [inline javascript/javascript1/week2/exerciseStep3Week2.js:62]: This works perfectly, but if you want to take your coding skills to the next level, might I introduce you to [guard clauses](https://learningactors.com/javascript-guard-clauses-how-you-can-refactor-conditional-logic/#:~:text=The%20guard%20clause,are%20met%2C%20or%20not%20met.&text=Guard%20clauses%20can%20reduce%20the,certain%20code%20to%20be%20executed.) Basically it is a way of doing away with so many levels of nested `if` statements and makes your code much simpler to read. Check them out and if it not clear, let's talk about them in the next class.

File: javascript/javascript1/week2/exerciseStep3Week2.js (hunk 2 above)
Comment (hamedveloper) [reply javascript/javascript1/week2/exerciseStep3Week2.js]: Oh , yes. Of course. I have added changes now.

Comment (hamedveloper) [reply javascript/javascript1/week2/exerciseStep3Week2.js]: I do not know if you can see changes or not!!I get confused as I add and commit changes and push it again in this place!!!!

**Pull request merged**